builders/build/substrate/adding-built-in-pallet.md

```diff
@@ -0,0 +1,192 @@
+
+What sets Substrate apart is its modular architecture, which enables the seamless integration of pre-built modules and the creation of custom ones, facilitating the development of blockchain protocols. 
+
+If the use case requires only EVM (Ethereum virtual machine) compatibility, then the provided template included in the [Tanssi repository](https://github.com/moondance-labs/tanssi#container-chain-templates){target=_blank} will meet the requirements and need no additional changes, but, teams willing to build a Substrate Appchain will need to add and compose the built-in modules and the custom-made ones into the runtime, compile and generate the chain specification and, finally, deploying through Tanssi protocol to evolve into a live ContainerChain.
+
+In this article, the necessary steps involved when adding a built-in module are covered.
+
+## Adding a Built-in Module to the Runtime {: #adding-a-built-in-module }
+
+As introduced in the [modularity](/learn/framework/modules) article, the Substrate framework already includes many built-in modules addressing a wide range of functionalities ready to use in your runtime.
+
+Modules are meant to provide the functionality needed in very different use cases such as DeFi, NFTs, or any other, and, therefore, they are basic building blocks inherently abstract that can be configured according to the specific needs of the Appchain.
+
+To add a module, the following steps are necessary:
+
+1. Make the dependency available within the project by declaring it in [Cargo](https://doc.rust-lang.org/cargo/){target=_blank}, the Rust language package manager
+2. Make the standard (`std`) features of the module available to the compiler
+3. Configure the module
+4. Add the module to the runtime
+5. Add default configuration in the chain specification
+
+In the following example, the very popular Substrate module `pallet-assets` will be added to the runtime of the provided EVM template, found in the [Tanssi repository](https://github.com/moondance-labs/tanssi){target=_blank}, specifically in the folder `container-chains/templates/frontier/`.
+
+### Declare the dependency {: #declare-dependency }
+
+Every package contains a manifest file named `Cargo.toml` stating, among other things, all the dependencies the package relies on, and the ContainerChain runtime is no exception. 
+
+To declare the dependency and make it available to the runtime, open the `Cargo.toml` file located in the folder `runtime` with a text editor and add the module, referencing the code in the official repository of the Polkadot SDK:
+
+```toml
+[dependencies]
+...
+pallet-assets = { git = "https://github.com/paritytech/polkadot-sdk", branch = "master", default-features = false }
```

Comment:
is there a toml formatter for this so it isn't so long? i like to try to avoid having code snippets with horizontal scrollbars

Reply:
toml format loves long lines :)